snmp/tests/test_profiles.py

```diff
@@ -2175,3 +2183,179 @@ def test_cisco_catalyst(aggregator):
     aggregator.assert_metric('snmp.sysUpTimeInstance', count=1)
     aggregator.assert_metric('snmp.devices_monitored', count=1)
     aggregator.assert_all_metrics_covered()
+
+
+@pytest.mark.usefixtures("dd_environment")
+def test_juniper_ex(aggregator):
+    run_profile_check('juniper-ex')
+    common_tags = common.CHECK_TAGS + [
+        'snmp_profile:juniper-ex',
+        'device_vendor:juniper-networks',
+    ]
+    _check_juniper_virtual_chassis(aggregator, common_tags)
+    _check_juniper_dcu(aggregator, common_tags)
+    _check_juniper_cos(aggregator, common_tags)
+    _check_juniper_firewall(aggregator, common_tags)
+    aggregator.assert_metric('snmp.devices_monitored', count=1)
+    aggregator.assert_all_metrics_covered()
```

Comment:
```suggestion
    aggregator.assert_all_metrics_covered()
    aggregator.assert_metrics_using_metadata(get_metadata_metrics(), check_submission_type=True)
```
Can we add this assertion? Then we can be sure the `metadata.csv` types match what is actually sent